CRM/Core/BAO/CustomQuery.php

```diff
@@ -387,20 +387,27 @@ public function where() {
 
               // CRM-14563,CRM-16575 : Special handling of multi-select custom fields
               if ($isSerialized && !empty($value) && !strstr($op, 'NULL') && !strstr($op, 'LIKE')) {
+                $sp = CRM_Core_DAO::VALUE_SEPARATOR;
                 if (strstr($op, 'IN')) {
-                  $value = str_replace(",", "[[:cntrl:]]*|[[:cntrl:]]*", $value);
+                  $value = str_replace(",", "$sp|$sp", $value);
                   $value = str_replace('(', '[[.left-parenthesis.]]', $value);
                   $value = str_replace(')', '[[.right-parenthesis.]]', $value);
                 }
                 $op = (strstr($op, '!') || strstr($op, 'NOT')) ? 'NOT RLIKE' : 'RLIKE';
-                $value = "[[:cntrl:]]*" . $value . "[[:cntrl:]]*";
+                $value = $sp . $value . $sp;
                 if (!$wildcard) {
```

Comment:
From Advanced Search, the values will always be the exact ones and will match without needing [[:cntrl:]]  With the current expression if you have values of "Ctrl-1 stuff" and "Ctrl-2 stuff" you cannot distinguish between them.
If we want to allow substring matches via api/Search Builder, maybe we should use an explicit syntax instead of just ignoring leading and trailing control characters?